src/olympia/activity/utils.py

```diff
@@ -0,0 +1,210 @@
+                    # Found the plain text part.
+                    payload = part.get_payload()
+                    break
+
+        # Decode quoted-printable data and remove non-breaking spaces.
+        payload = (quopri.decodestring(payload)
+                         .replace('\xc2\xa0', ' '))
+        payload = self.extra_email_reply_parse(payload)
+        self.reply_text = EmailReplyParser.read(payload).reply
+
+    def extra_email_reply_parse(self, email):
+        """
+        Adds an extra case to the email reply parser where the reply is
+        followed by headers like "From: amo-editors@mozilla.org" and
+        strips that part out.
+        """
+        email_header_re = re.compile('From: [^@]+@[^@]+\.[^@]+')
+        split_email = email_header_re.split(email)
+        if split_email[0].startswith('From: '):
+            # In case, it's a bottom reply, return everything.
+            return email
+        else:
+            # Else just return the email reply portion.
+            return split_email[0]
+
+    def _get_address_line(self):
+        return parseaddr(self.email['to']) or parseaddr(self.email(['reply']))
+
+    def get_uuid(self):
+        name, addr = self._get_address_line()
+
+        if addr.startswith(self.address_prefix):
+            # Strip everything between "reviewreply+" and the "@" sign.
+            uuid = addr[len(self.address_prefix):].split('@')[0]
+        else:
+            log.info('TO: address missing or not related to activity emails. '
+                     '(%s)' % unicode(self.email).strip())
+            return False
+
+        return uuid
+
+    def get_body(self):
+        return self.reply_text
+
+
+def add_email_to_activity_log(email_text):
+    log.debug("Saving from email reply")
+
+    parser = ActivityEmailParser(email_text)
+    if hasattr(parser, 'decode_error'):
+        return False
+
+    uuid = parser.get_uuid()
+
+    if not uuid:
+        return False
+    try:
+        token = ActivityLogToken.objects.get(uuid=uuid)
+    except ActivityLogToken.DoesNotExist:
+        log.error('An email was skipped with non-existing uuid %s.' % uuid)
+        return False
+
+    version = token.version
+    user = token.user
+    if token.is_valid():
+        log_type = None
+
+        if (version.addon.authors.filter(pk=user.pk).exists()):
+            log_type = amo.LOG.DEVELOPER_REPLY_VERSION
+        elif ((version.addon.is_listed and
+               acl.action_allowed_user(user, 'Addons', 'Review')) or
+              (not version.addon.is_listed and
+               acl.action_allowed_user(user, 'Addons', 'ReviewUnlisted'))):
+            log_type = amo.LOG.REVIEWER_REPLY_VERSION
+
+        if log_type:
+            note = log_and_notify(log_type, parser.get_body(), user, version)
+            log.info('A new note has been created (from %s using tokenid %s).'
+                     % (user.id, uuid))
+            token.increment_use()
```

Comment:
There is no `save()` after this so it looks like the new `use_count` never makes it to the database? On a related note, I could not find a test covering this (had there been one it would probably have exposed this problem)